Cannot create new project with npm 9, peer violations from dependencies

```
❯ npm -v
9.3.0

~/Development/tmp 
❯ node -v
v18.16.0

❯  npx ember-cli@v5.0.0-beta.0 new my-app
# ...
Command failed with exit code 1: npm install --loglevel error
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: my-app@0.0.0
npm ERR! Found: ember-cli@5.0.0-beta.0
npm ERR! node_modules/ember-cli
npm ERR!   dev ember-cli@"~5.0.0-beta.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer ember-cli@"^3.2.0 || ^4.0.0" from ember-cli-dependency-checker@3.3.1
npm ERR! node_modules/ember-cli-dependency-checker
npm ERR!   dev ember-cli-dependency-checker@"^3.3.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/nullvoxpopuli/.npm/_logs/2023-06-02T21_44_53_662Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/nullvoxpopuli/.npm/_logs/2023-06-02T21_44_53_662Z-debug-0.log


Stack Trace and Error Report: /tmp/error.dump.5c1ae3f0d0d273901322b8f9715ec704.log

```

in this particular error, `ember-cli-dependency-checker` is not declaring support for ember-cli@v5

--- a/package.json
+++ b/package.json
@@ -52,7 +52,7 @@
     "sinon": "^13.0.2"
   },
   "peerDependencies": {
-    "ember-cli": "^3.2.0 || ^4.0.0 || ^5.0.0"
+    "ember-cli": "^3.2.0 || >=4.0.0"
   },
   "engines": {
     "node": ">= 6"

--- a/package.json
+++ b/package.json
@@ -52,7 +52,7 @@
     "sinon": "^13.0.2"
   },
   "peerDependencies": {
-    "ember-cli": "^3.2.0 || ^4.0.0"
+    "ember-cli": "^3.2.0 || ^4.0.0 || ^5.0.0"
   },
   "engines": {
     "node": ">= 6"

